lib/web_ui/lib/src/engine/compositor/picture_recorder.dart

```diff
@@ -2,32 +2,32 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-// @dart = 2.6
+
 part of engine;
 
 class SkPictureRecorder implements ui.PictureRecorder {
-  ui.Rect _cullRect;
-  js.JsObject _recorder;
-  SkCanvas _recordingCanvas;
+  ui.Rect? _cullRect;
+  js.JsObject? _recorder;
+  SkCanvas? _recordingCanvas;
 
-  SkCanvas beginRecording(ui.Rect bounds) {
+  SkCanvas? beginRecording(ui.Rect bounds) {
     _cullRect = bounds;
     _recorder = js.JsObject(canvasKit['SkPictureRecorder']);
     final js.JsObject skRect = js.JsObject(canvasKit['LTRBRect'],
         <double>[bounds.left, bounds.top, bounds.right, bounds.bottom]);
-    final js.JsObject/*!*/ skCanvas =
-        _recorder.callMethod('beginRecording', <js.JsObject>[skRect]);
+    final js.JsObject skCanvas =
+        _recorder!.callMethod('beginRecording', <js.JsObject>[skRect]);
```

Comment:
how come `skCanvas` can be non-nullable here?

Reply:
According to (abbreviated) implementation:

```cpp
SkCanvas* SkPictureRecorder::beginRecording(const SkRect& userCullRect,
                                            sk_sp<SkBBoxHierarchy> bbh,
                                            uint32_t recordFlags /* = 0 */) {
    ... omitted code ...
    fActivelyRecording = true;
    return this->getRecordingCanvas();
}

SkCanvas* SkPictureRecorder::getRecordingCanvas() {
    return fActivelyRecording ? fRecorder.get() : nullptr;
}
```

Notice that `beginRecording` always does `fActivelyRecording = true;` prior to calling to `getRecordingCanvas`, so it will never return null.